libpromises/eval_context.c

```diff
@@ -1865,10 +1871,7 @@ VariableTableIterator *EvalContextVariableTableFromRefIteratorNew(const EvalCont
 
 const void *EvalContextVariableControlCommonGet(const EvalContext *ctx, CommonControl lval)
 {
-    if (lval >= COMMON_CONTROL_MAX)
-    {
-        return NULL;
-    }
+    assert(lval >= 0 && lval < COMMON_CONTROL_MAX);
```

Comment:
If it triggers we'll know this from the debug builds. I didn't find a reason for this check to be in production, the debug build should easily catch a programming mistake (famous last words)...